notion-server/src/ethereum/mod.rs

```diff
@@ -551,17 +549,15 @@ impl TryFrom<&args::ShardTransaction> for ShardTransaction {
 
 #[derive(Default, Debug)]
 struct BeaconBlock {
-    cross_links: HashMap<ShardChainIndex, CrossLink>,
+    cross_links: Vec<CrossLink>,
```

Comment:
It is my understanding that it is possible to have a beacon block that is missing a cross link for a particular shard, which is why I recommended a `Vec<Option<CrossLink>>`.

Is your intent to represent a missing crosslink with an empty map inside the `CrossLink` struct?

Reply:
You are correct, I'll need to update